src/wallet/mod.rs

```diff
@@ -316,6 +316,51 @@ where
         }
     }
 
+    /// Ensures that there are at least `max_addresses` or `CACHE_ADDR_BATCH_SIZE` addresses
+    /// cached in the database if the descriptor is derivable, or 1 address if it is not.
+    /// Will return `Ok(true)` if there are new addresses generated (either external or internal),
+    /// and `Ok(false)` if all the required addresses are already cached. This function is used internally
+    /// by [`Wallet::sync`] in wallets that have a [`Blockchain`] client, and is useful for direct
```

Comment:
Let's not talk about what Wallet::sync does internally since this doc could easily become outdated. After #535 is merged all wallets will be "offline" and can `sync` if/when a blockchain becomes available. So I think this sentence is handy to remove.

It is worth mentioning that the reason you do this is so that `is_mine` works.